MapboxCoreNavigation/OfflineDirections.swift

```diff
@@ -196,6 +203,13 @@ public class NavigationDirections: Directions {
         
         NavigationDirectionsConstants.offlineSerialQueue.async { [weak self] in
             guard let result = self?.navigator.getRouteForDirectionsUri(url.absoluteString) else {
+                DispatchQueue.main.async {
+                    completionHandler(session, .failure(.cancelled))
+                }
+                return
+            }
+            
+            guard result.isSuccess else {
                 DispatchQueue.main.async {
                     completionHandler(session, .failure(.noData))
```

Comment:
Seem like `.noData` is not the correct option here. `result` has only 2 properties exposed: `isSuccess` (which is so far was unused) and `json` (which may contain a valid response or an error message). In current implementation, I struggle to find a correct definition of "router returned no data" case to fire `.noData` message.

Reply:
This return value is a holdover from 0486778621f774fa88289617dd0c18b12d227ab6. It might be fine to reuse `.noData` for the `.cancelled` case. In the event of a failure, there should be an error that we can pass along.